server/api/posts/repository.go

```diff
@@ -29,3 +33,79 @@ func (repository *PostRepository) CreatePost(post model.Post, authorId uint) err
 
 	return repository.database.Exec(query, post.Title, post.Slug, post.Content, post.Status, authorId, createdAt).Error
 }
+
+// FUNCTION TO GET ALL POSTS
+func (reposistory *PostRepository) GetAllPosts() (error, []model.Post) {
+	fmt.Println("STARTING GET/posts request: ")
+	var err error
+	var catID sql.NullInt64
```

Comment:
what is `catId`?

Reply:
catID is categoryID with type sql.NullInt64 , which allow to pass the null value in the posts table because *uint in struct cannot handler this null value